web/src/stream_list.ts

```diff
@@ -943,7 +943,20 @@ export function set_event_handlers({
                 false,
                 (topic_names: string[]) => topic_names,
             );
-            const topic_item = topic_list_info.items[0];
+            let topic_item = topic_list_info.items[0];
+
+            if (
+                user_settings.web_channel_default_view ===
+                web_channel_default_view_values.top_unread_topic_in_channel.code
+            ) {
+                for (const topic_list_item of topic_list_info.items) {
+                    if (unread.topic_has_any_unread(stream_id, topic_list_item.topic_name)) {
+                        topic_item = topic_list_item;
+                        break;
+                    }
+                }
+            }
```

Comment:
This logic looks correct for most topics. How does this work for muted topics? I think we want explicitly muted topics to not be prioritized over "the most recent normal topic in the channel"